### ISSUE TITLE
What happens when FETCH_HEADER is lost?

### ISSUE THREAD

**Issue by afrind** [moq-wg/moq-transport#632]
I have the following test case where the publisher makes an API error and publishes an object on a fetch stream with the wrong length:

C->S: FETCH
S->C: FETCH_OK
S->C: open fetch stream (2), 
S->C: write FETCH_HEADER (2, delayed/lost)
S: API error, RESET_STREAM (2)

The client is expecting a fetch stream that will never come.  

The client needs to have a timer on the arrival of the FETCH_HEADER in any case, because a malicious or buggy server might never send one on purpose.  We should add text to this effect.

We can also short circuit the timer in cases like the above by using RESET_STREAM_AT (always deliver the FETCH_HEADER).

**Comment by vasilvv** [comment]
I feel like between this and the other issue you filed, this makes a good case for switching fetches to bidi streams.

**Comment by afrind** [comment]
Individual Comment:

I was thinking a bit about N control streams.  The main argument against it has been ordering would not be guaranteed for important sequences of operations, but I think that could be handled with a control message sequence number.  I guess that opens a new flow control problem though.

**Comment by vasilvv** [comment]
I'm not sure what N control streams would do to help the state tracking problem.

**Comment by afrind** [comment]
Individual Comment:

Sorry, that was a tangent -- FETCH is a control message, so putting it on a bidi stream implicitly makes N control streams.  Once we do that, it's tempting to put, say, SUBSCRIBE/SUBSCRIBE_OK/SUBSCRIBE_ERROR/SUBSCRIBE_UPDATE for a given Subscribe ID also on a bidi stream, as well as other message groups (ANNOUNCE*, SUBSCRIBE_ANNOUNCES*).

**Comment by vasilvv** [comment]
That is one possible direction, though I'm not sure that it makes much sense to me.  A fetch is well-suited for this since everything can happen on a single stream; subscribes span multiple streams, making the advantage of simplifying the state management much less appealing.

**Comment by ianswett** [comment]
I agree with Victor that the case for FETCH on a single stream is much cleaner than SUBSCRIBE.

**Comment by afrind** [comment]
Individual Comment: 

If I recall the design principles of FETCH from Boston, I believe it was intentional to allow an intermediary to begin sending objects before sending FETCH_OK.  This is because FETCH_OK contains information about liveness of the track/live head, which may require an RTT upstream.  If we move FETCH to a bidi-stream, how would this change?  

1. Keep FETCH_OK on control stream but FETCH and Objects go on the bidi stream
2. Put FETCH, FETCH_OK and Objects on the bidi stream, but make the wire format flexible so FETCH_OK can appear mid-stream.
3. Put FETCH, FETCH_OK and Objects on the bidi stream, but require the FETCH_OK to come first

**Comment by afrind** [comment]
I'm disinclined to keep this issue tracking using a bidirectional stream for FETCH.  It solves some problems (guaranteed return stream) but creates others (see https://github.com/moq-wg/moq-transport/issues/632#issuecomment-2605450825, and also would break Joining FETCH).

I think it's sufficient to add editorial text indicating that receivers need to have a timeout for the data stream for a FETCH to arrive, and indicate that RESET_STREAM_AT offers advantages to the receiver in this regard.

@vasilvv or @ianswett if you want to change fetch to a bidi stream, please open a separate issue.
